src/IK/src/GravityTask.cpp

```diff
@@ -0,0 +1,228 @@
+        return false;
+    }
+
+    std::string robotVelocityVariableName;
+    if (!ptr->getParameter("robot_velocity_variable_name", m_robotVelocityVariable.name))
+    {
+        log()->error("{} [{}] while retrieving the robot velocity variable.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    // set the gains for the controllers
+    if (!ptr->getParameter("kp", m_kp))
+    {
+        log()->error("{} [{}] to get the proportional linear gain.", errorPrefix, m_description);
+        return false;
+    }
+
+    // set the base frame name
+    if (!ptr->getParameter("base_name", m_baseName))
+    {
+        log()->debug("{} [{}] to get the base name. Using default \"\"",
+                     errorPrefix,
+                     m_description);
+    }
+
+    if (m_baseName != "")
+    {
+        m_baseIndex = m_kinDyn->getFrameIndex(m_baseName);
+
+        if (m_baseIndex == iDynTree::FRAME_INVALID_INDEX)
+            return false;
+    }
+
+    // set the finger tip frame Name
+    if (!ptr->getParameter("target_frame_name", m_targetFrameName)) 
+    {
+        log()->error("{} [{}] to get the end effector frame name.", errorPrefix, m_description);
+        return false;
+    }
+
+    m_targetFrameIndex = m_kinDyn->getFrameIndex(m_targetFrameName);
+
+    if (m_targetFrameIndex == iDynTree::FRAME_INVALID_INDEX)
+        return false;
+    
+    // here you need to get the indexes of the frames and check that they exists
+
+    m_isInitialized = true;
+
+    return true;
+}
+
+bool GravityTask::update()
+{
+    using namespace BipedalLocomotion::Conversions;
+    using namespace iDynTree;
+
+    m_isValid = false;
+
+    // Normalize Accelerometer vector
+    m_accDenomNorm = sqrt(pow(m_currentAcc(0),2) + pow(m_currentAcc(1),2) + pow(m_currentAcc(2),2));
+
+    for (size_t i = 0; i < m_currentAcc.size(); i++)
+    {
+        m_currentAccNorm(i) = m_currentAcc(i) / m_accDenomNorm;
+    }
+    
+    if (m_baseName == "")
+    {
+            // get the jacobian
+        if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
+        {
+            log()->error("[GravityTask::update] Unable to get the jacobian.");
+            return m_isValid;
+        }
+
+    }
+    else
+    {
+            // get the relative jacobian
+        if (!m_kinDyn->getRelativeJacobian(m_baseIndex, m_targetFrameIndex, m_relativeJacobian))
+        {
+            log()->error("[GravityTask::update] Unable to get the relative jacobian.");
+            return m_isValid;
+        }
+        m_jacobian.bottomRightCorner(3, m_kinDyn->getNrOfDegreesOfFreedom()) = m_relativeJacobian;
+
+    }
+    
+    m_A.resize(2, 6 + m_kinDyn->getNrOfDegreesOfFreedom()); //the jacobian matrix 1x(6+ndofs)
+    m_A = m_Am * m_relativeJacobian + (0.001 * Eigen::MatrixXd::Ones(1, 6 + m_kinDyn->getNrOfDegreesOfFreedom()));
```

Comment:
I think you need to use ``m_jacobian`` here. Also I think there is a problem with the dimensions:
- ``m_Am`` is 2 x 3
- ``m_relativeJacobian`` is 6xn
-  ``Eigen::MatrixXd::Ones(1, 6 + m_kinDyn->getNrOfDegreesOfFreedom())`` is 1x(6+n)

The output matrix is supposed to be a 2x(6+n)

Reply:
FIxed. (I hope so 😄)